include/deal.II/fe/fe_values.h

```diff
@@ -4588,6 +4590,110 @@ class FESubfaceValues : public FEFaceValuesBase<dim, spacedim>
 };
 
 
+/**
+ * Finite element evaluated in the quadrature points of an
+ * ImmersedSurfaceQuadrature of a cell.
+ *
+ * The shape functions values and their derivatives are the same as for an
+ * FEValues-object, but the JxW-values are computed with the transformation
+ * described in the documentation of ImmersedSurfaceQuadrature. Further, the
+ * normal_vector-function returns the normal to the immersed surface.
+ *
+ * The reinit-function of this class exist mostly to be consistent with the
+ * other FEValues-like classes. The immersed quadrature rule will typically vary
+ * between each cell of the triangulation. Thus, an FEImmersedSurfaceValues
+ * object can, typically, not be reused for different cells.
+ *
+ * See also documentation in FEValuesBase.
+ *
+ * @ingroup feaccess
+ */
+template <int dim>
+class FEImmersedSurfaceValues : public FEValuesBase<dim, dim>
```

Comment:
Yes, I think it would make sense to keep things collected in that namespace for compactness. `fe_values.h` is a very large file already, so it just helps with the additional implementation. Since we do not put restrictions on the include files between different subfolders in deal.II, it is also no problem to load the additional header in the mapping classes and set the appropriate forward declaration for the `friends` specification.